frontend/public/components/monitoring/alert-manager-config.tsx

```diff
@@ -219,23 +227,24 @@ const Receivers: React.FC<AlertManagerProps> = ({ config }) => {
 };
 
 const AlertManagerConfiguration: React.FC<AlertManagerConfigurationProps> = ({ obj: secret }) => {
-  const alertManagerYaml = _.get(secret, ['data', 'alertmanager.yaml']);
-  let errorMsg;
-  let yamlStringData;
-  let config;
-
-  if (!_.isEmpty(alertManagerYaml)) {
-    yamlStringData = Base64.decode(alertManagerYaml);
-    try {
-      config = safeLoad(yamlStringData);
-    } catch (e) {
-      errorMsg = `Error parsing YAML: ${e}`;
-    }
+  const [errorMsg, setErrorMsg] = React.useState('');
+  let config: AlertManagerConfig;
+  if (!errorMsg) {
+    config = getAlertManagerConfig(secret, setErrorMsg);
   }
 
   return (
-    <div className="co-m-pane__body">
-      {errorMsg && <span>{errorMsg}</span>}
+    <div>
+      {errorMsg && (
+        <Alert
```

Comment:
Probably a little more readable with an early return like
```
if (errorMsg) {
  return <Alert ...
}
```
to remove the need for the wrapping `<div>` and the `!errorMsg` check.

Reply:
fixed, see last commi

Reply:
👍